stmtools/_io.py

```diff
@@ -0,0 +1,137 @@
+
+    `from_csv` does not retrieve time stamps based on column names. The `time` coordinate of 
+      the output STM will be a monotonic integer series starting from 0.
+
+    Args:
+        file (str | Path): Path to the csv file.
+        space_pattern (str, optional): RE pattern to match space attribute columns. 
+          Defaults to "^pnt_".
+        spacetime_pattern (dict | None, optional): A dictionay mapping RE patterns of each 
+          space-time attribute to corresponding variable names. Defaults to None, which means
+          the following map will be applied: 
+          {"^d_": "deformation", "^a_": "amplitude", "^h2ph_": "h2ph"}.
+        coords_cols (list | dict, optional): List of columns to be used as space coordinates. 
+          When `coords_cols` is a dictionary, a reaming will be performed per coordinates. 
+          Defaults to None, then the following renaming will be performed: 
+          "{"pnt_lat": "lat", "pnt_lon": "lon"}"
+        output_chunksize (dict | None, optional): Chunksize of the output. Defaults to None, 
+          then the size of the first chunk in the DaskDataFrame will be used, up-rounding to 
+          the next 5000.
+        blocksize (int | str | None, optional): Blocksize to load the csv. 
+          Defaults to 200e6 (in bytes). See the documentation of 
+          [dask.dataframe.read_csv](https://docs.dask.org/en/stable/generated/dask.dataframe.read_csv.html)
+
+    Returns:
+        xr.Dataset: Output STM instance
+    """
+
+    # Load csv as Dask DataFrame
+    ddf = dd.read_csv(file, blocksize=blocksize)
+
+    # Take the first column and compute the chunk sizes
+    da_col0 = ddf[ddf.columns[0]].to_dask_array()
+    da_col0.compute_chunk_sizes()
+    chunks = da_col0.chunks
+
+    # Count time dimension by one spacetime column
+    if spacetime_pattern is None:
+        spacetime_pattern = {"^d_": "deformation", "^a_": "amplitude", "^h2ph_": "h2ph"}
+    time_shape = 0
+    if spacetime_pattern is not None:
+        key = list(spacetime_pattern.keys())[0]
+        for column in ddf.columns:
+            if re.match(re.compile(key), column):
+                time_shape += 1
+
+    # Initiate a template STM
+    coords = {
+        "space": range(da_col0.shape[0]),
+        "time": range(time_shape),
+    }
+    stmat = xr.Dataset(coords=coords)
+
+    # Read csv by columns
+    dict_temp_da = dict()  # Temporary dask array collector
+    for k in spacetime_pattern.keys():
+        dict_temp_da[k] = []
+
+    # Temporaly save time-series columns to lists in dict_temp_da
+    for column in ddf.columns:
+        if _is_space(space_pattern, column):
+            da_pnt = ddf[column].to_dask_array(lengths=chunks[0])
+            stmat = stmat.assign({column: (("space"), da_pnt)})
+        else:
+            for k in spacetime_pattern.keys():
+                if re.match(re.compile(f"{k}"), column):
```

Comment:
the same comment as above, let's add `else` statement with a clear error message to help users if the pattern in csv file is not valid.